explorer/interpreter/interpreter.cpp

```diff
@@ -1278,36 +1309,52 @@ auto Interpreter::StepExp() -> ErrorOr<Success> {
     case ExpressionKind::CompoundMemberAccessExpression: {
       const auto& access = cast<CompoundMemberAccessExpression>(exp);
       bool forming_member_name = isa<TypeOfMemberName>(&access.static_type());
+
       if (act.pos() == 0) {
-        // First, evaluate the first operand.
+        // First evaluate the first operand
         if (access.is_addr_me_method()) {
           return todo_.Spawn(std::make_unique<LValAction>(&access.object()));
         } else {
           return todo_.Spawn(
               std::make_unique<ExpressionAction>(&access.object()));
         }
-      } else if (act.pos() == 1 && access.impl().has_value() &&
-                 !forming_member_name) {
+      } else if ((act.pos() == 1 && access.impl().has_value() &&
```

Comment:
I'm not sure that flow makes sense anymore if we end up diverging for all these subcases. The logic becomes difficult to understand. Would that look better if we invert the "step" and "target" conditions?

For example by doing (pseudocode)
```
  else {
    if (access.impl().has_value() && !forming_member_name) {
       if (act.pos() == 1) {
       ...
       } else if (act.pos() == 2) {
       ...
       } else {
          return ...;
       }
    else if (found_in_interface=...; found_in_interface) {
       // Pos 1-n
    } else if () {
      ...
    }
    }
  }
```

Reply:
Yes I tried it and certainly makes the code more readable (maybe just for me as I implemented it) but an extra level of nesting it comes with.